ui/src/js/directives.js

```diff
@@ -822,4 +822,32 @@
             }
         };
     });
+
+    widgetModule.directive('motechFileUpload', ['$compile', '$timeout', '$http', '$templateCache', function ($compile, $timeout, $http, $templateCache) {
+        var templateLoader;
+
+        return {
+            restrict: 'E',
+            replace : true,
+            transclude: true,
+            compile: function (tElement, tAttrs, scope) {
+                var url = '../server/resources/partials/motech-file-upload.html',
+
+                templateLoader = $http.get(url, {cache: $templateCache})
+                    .success(function (html) {
+                        tElement.html(html);
+                    });
+
+                return function (scope, element, attrs) {
+                    templateLoader.then(function () {
+                        element.html($compile(tElement.html())(scope));
+                        var input = element.find("input");
+                        if (attrs.accept !== null &&  attrs.accept !== "") {
```

Comment:
what about undefined ? the best way is to check it like the following : 'if (attrs.accept)'